### ISSUE TITLE
Long cronjob names can lead to issues on deplyoments

### ISSUE THREAD

**Issue by dasrecht** [uselagoon/build-deploy-tool#185]

**Comment by sjerdo** [comment]
We also experience this issue for long cronjob names:

```
The CronJob "cronjob-cli-cronjobName" is invalid: metadata.name: Invalid value: "cronjob-cli-cronjobName": must be no more than 52 characters
```
* Replaced the cronjob name with value `cronjobName`

**Comment by shreddedbacon** [comment]
> Can you cap these names to 52 characters? The length depends on the container and cronjob name

Yes, we can probably truncate them and add a hash so the names remain unique

**Comment by shreddedbacon** [comment]
I'm already looking at template generation functions, so I'll flag this as something to resolve in that part of work.

Obviously the short term workaround is to choose a shorter name, but obviously not ideal.
